src/rgw/rgw_process.cc

```diff
@@ -466,7 +466,7 @@ int process_request(const RGWProcessEnv& penv,
     rgw_log_op(rest, s, op, penv.olog.get());
   }
 
-  if (op) {
+  if (op && op->get_type() != RGW_OP_PUT_OBJ) {
```

Comment:
would prefer to avoid special cases for op type here (same feedback as https://github.com/ceph/ceph/pull/63244#discussion_r2098636042 for complete multipart)

p.s. this pr will conflict with https://github.com/ceph/ceph/pull/63244

Reply:
once this is merged. I will update the other PR with the new `always_log()` API